vault/map.jinja

```diff
@@ -1,11 +1,15 @@
-{% import_yaml "vault/defaults.yaml" as defaults %}
-{% import_yaml "vault/osfamilymap.yaml" as osfamilymap %}
+# -*- coding: utf-8 -*-
+# vim: ft=sls syntax=yaml softtabstop=2 tabstop=2 shiftwidth=2 expandtab autoindent
+
+{% import_yaml "vault/yaml/defaults.yaml" as defaults %}
+{% import_yaml "vault/yaml/osfamilymap.yaml" as osfamilymap %}
+{% import_yaml "vault/yaml/initfamilymap.yaml" as initfamilymap %}
 
 {% set vault = salt['grains.filter_by'](
-    defaults,
-    merge=salt['grains.filter_by'](
-      osfamilymap,
-      merge=salt['pillar.get']('vault', {}),
-    ),
-    base='vault')
- %}
+  defaults, merge=salt['grains.filter_by'](
+    osfamilymap, merge=salt['grains.filter_by'](
+      initfamilymap, grain='init', merge=salt['pillar.get']('vault', {}),
+      base='vault'),
+    base='vault'),
+  base='vault')
+%}
```

Comment:
This could be tidied up a little more with reference to https://github.com/saltstack-formulas/template-formula/blob/master/template/map.jinja.

Reply:
To be more helpful:

1. Use `or {}`.
1. Consider the [comment below](https://github.com/saltstack-formulas/vault-formula/pull/23/files?file-filters%5B%5D=.sls&file-filters%5B%5D=.yaml&hide-deleted-files=true#r268416081).
1. Merging strategy and layout.

Reply:
Thanks for the updated comment but I still don't follow. If I remove the `base` part it doesn't merge correctly. Could you please clarify?

Reply:
@rbjorklin Use this instead:

```jinja
# -*- coding: utf-8 -*-
# vim: ft=sls syntax=yaml softtabstop=2 tabstop=2 shiftwidth=2 expandtab autoindent

{% import_yaml "vault/yaml/defaults.yaml" or {} as default_settings %}
{% import_yaml "vault/yaml/osfamilymap.yaml" or {} as osfamilymap %}
{% import_yaml "vault/yaml/initfamilymap.yaml" or {} as initfamilymap %}

{%- set defaults = salt['grains.filter_by'](default_settings,
    default='vault',
    merge=salt['grains.filter_by'](osfamilymap, grain='os_family',
      merge=salt['grains.filter_by'](initfamilymap, grain='init',
        merge=salt['pillar.get']('vault:lookup', default={})
      )
    )
) %}

{#- Merge the vault pillar #}
{%- set vault = salt['pillar.get']('vault', default=defaults, merge=True) %}
```

Please test the above since it's a direct conversion using the `template-formula` as a basis.

Reply:
@rbjorklin Since you've just reset the location of the `.yaml` files, then the paths need to be updated for the example above (`vault/yaml/...` => `vault/...`).

Reply:
What's the `vault:lookup` for?

Reply:
@rbjorklin The `lookup` is mentioned in the main documentation both [here](https://docs.saltstack.com/en/latest/topics/development/conventions/formulas.html#abstracting-static-defaults-into-a-lookup-table) and [here](https://docs.saltstack.com/en/latest/topics/best_practices.html).  For some discussions about this, refer to:

* https://github.com/saltstack-formulas/redis-formula/issues/75#issuecomment-459451176
  - The discussion starts with this comment.
* https://github.com/saltstack-formulas/apache-formula/pull/252#issue-248307707

While `vault:lookup` may not exist now, this structure allows for it to be adopted at a later stage.

Reply:
Cleaned up!

Reply:
@rbjorklin Thanks for this.  Just missing the `or {}`, which was raised as an issue [here](https://github.com/saltstack-formulas/template-formula/issues/59) and was fixed [here](https://github.com/saltstack-formulas/template-formula/pull/60).  So that would be:

```sls
{% import_yaml "vault/defaults.yaml" or {} as defaults %}
{% import_yaml "vault/osfamilymap.yaml" or {} as osfamilymap %}
{% import_yaml "vault/initfamilymap.yaml" or {} as initfamilymap %}
```